app/models/user_stat.rb

```diff
@@ -175,19 +175,27 @@ def self.update_view_counts(last_seen = 1.hour.ago)
 
     # Update denormalized posts_read_count
     DB.exec(<<~SQL, seen_at: last_seen)
+      WITH filtered_users AS (
+        SELECT id FROM users u
+        JOIN user_stats ON user_id = u.id
+        WHERE last_seen_at > now() - interval '1 hour'
+        AND posts_read_count < 10000
+      ),
+      filtered_topics AS (
+         SELECT id FROM topics
+         WHERE archetype = 'regular'
+         AND deleted_at IS NULL
+      )
```

Comment:
What are the performance benefit of extracting this into a CTE because I'm not sure if this will have an impact.